packages/components/addon/components/hds/tabs/index.js

```diff
@@ -7,45 +7,71 @@ import Component from '@glimmer/component';
 import { tracked } from '@glimmer/tracking';
 import { action } from '@ember/object';
 import { assert } from '@ember/debug';
-import { schedule } from '@ember/runloop';
+import { next, schedule } from '@ember/runloop';
 
 export default class HdsTabsIndexComponent extends Component {
   @tracked tabNodes = [];
   @tracked tabIds = [];
   @tracked panelNodes = [];
   @tracked panelIds = [];
-  @tracked selectedTabIndex;
+  @tracked _selectedTabIndex = this.args.selectedTabIndex ?? 0;
+  @tracked selectedTabId;
+
+  get selectedTabIndex() {
+    if (this.args.selectedTabIndex !== undefined) {
+      return this.args.selectedTabIndex;
+    } else {
+      return this._selectedTabIndex;
+    }
+  }
+
+  set selectedTabIndex(value) {
+    if (this.args.selectedTabIndex) {
+      // noop
+    } else {
+      this._selectedTabIndex = value;
+    }
```

Comment:
do you think it reads easier if we swap this statement?

Reply:
I have changed the logic to make it more explicit and clear to understand (by introducing a `isControlled` tracked variable) - See commit https://github.com/hashicorp/design-system/pull/1688/commits/8fd10b9ac303b23fc534e3e295919df11c3a6e46